crates/hir/src/lib.rs

```diff
@@ -1294,13 +1294,17 @@ impl Local {
         self.name(db) == Some(name![self])
     }
 
+    pub fn is_mut_binding(self, db: &dyn HirDatabase) -> bool {
+        let body = db.body(self.parent.into());
+        matches!(&body[self.pat_id], Pat::Bind { mode: BindingAnnotation::Mutable, .. })
+    }
+
     pub fn is_mut(self, db: &dyn HirDatabase) -> bool {
```

Comment:
It's weird that we need two functions here. A doc comment at minimum is needed to explain the difference between the two

Reply:
But `let ref mut foo` local isn't actually mutable. As in, you can't reassign the `foo`. Which test break if you just fix the function to not consider `rem mut` as mutable? Those tests might want to look at the mutability of the type of the local instead.

Reply:
Oh ye nevermind the names were better before, `is_mut_binding` was meant for the binding being mutable I'll revert the names but stick the comments to it

Reply:
Ah, now I see your point. `is_mut` is solely used by highlighting which already checks the type so swapping that seems to work.